CoffeeChat/Controller/EditAvailabilityViewController.swift

```diff
@@ -0,0 +1,576 @@
+    private var editScrollView: UIScrollView!
+    private var locationsCollectionView: UICollectionView!
+    private let saveBarButtonItem = UIBarButtonItem()
+    private let scheduleLocationLabel = UILabel()
+    private let scheduleTimeLabel = UILabel()
+    private let scheduleLocationSubLabel = UILabel()
+    private var timeCollectionView: UICollectionView!
+    
+    // MARK: - Time CollectionView Sections
+    private struct TimeSection {
+        let type: SectionType
+        var items: [ItemType]
+    }
+
+    private enum SectionType: String {
+        case afternoon = "Afternoon"
+        case evening = "Evening"
+        case morning = "Morning"
+    }
+    
+    private enum ItemType {
+        case header(String)
+        case time(String)
+        
+        func getTime() -> String? {
+            switch self {
+            case .time(let time):
+                return time
+            default:
+                return nil
+            }
+        }
+    }
+    
+    private var timeSections: [TimeSection] = []
+    
+    // MARK: - Time Data Vars
+    private let timeInteritemSpacing: CGFloat = 4
+    private let sectionInsets = UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 8)
+
+    private let dayCellReuseId = "dayCellReuseIdentifier"
+    private let timeCellReuseId = "timeCellReuseIdentifier"
+    
+    // All possible times available for parts of a day
+    private var allAfternoonTimes = ["1:00", "1:30", "2:00", "2:30", "3:00", "3:30", "4:00", "4:30"]
+    private var allEveningTimes = ["5:00", "5:30", "6:00", "6:30", "7:00", "7:30", "8:00", "8:30"]
+    private var allMorningTimes = ["9:00", "9:30", "10:00", "10:30", "11:00", "11:30", "12:00", "12:30"]
+    
+    // For times presented to user for parts of a day
+    private var afternoonTimes: [String] = []
+    private var eveningTimes: [String] = []
+    private var morningTimes: [String] = []
+    
+    // For section items with a header and times
+    private var afternoonItems: [ItemType] = []
+    private var eveningItems: [ItemType] = []
+    private var morningItems: [ItemType] = []
+    
+    private var availabilities: [String: [String]] = [:]
+    private var daysAbbrev = ["Su", "M", "Tu", "W", "Th", "F", "Sa"]
+    private let daysDict = ["Su": "Sunday", "M": "Monday", "Tu": "Tuesday", "W": "Wednesday", "Th": "Thursday", "F": "Friday", "Sa": "Saturday"]
+    private var selectedDay: String = "Su"
+
+    // TODO: Change values after connecting to backend and get user's saved availabilities
+    private var savedAvailabilities: [String: [String]] = ["Monday": ["5:30", "6:00", "6:30"], "Wednesday": ["10:30", "11:00", "11:30", "2:00", "2:30",], "Friday": ["1:30", "2:00", "5:30", "6:00", "6:30"], "Saturday": ["7:30", "11:00", "11:30", "12:00", "12:30"]]
+
+    // MARK: - Location CollectionView Sections
+    private enum LocationSection {
+        case campus([String])
+        case ctown([String])
+    }
+    
+    // MARK: - Location Data Vars
```

Comment:
and same goes for looking into modularizing the location selection collection view into its own component